react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await readItems();
+        setItems(response);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
+      }
+    }
+    fetchData();
+  }, [error]);
```

Comment:
Why is `error` a dependency? It's not accessed anywhere in the useEffect.

Oh, you're doing this to get it to reload after an error. Clever, but setting isLoading as mentioned above is the most correct way to do it.